apollo-router/src/plugin/binary_plugins.rs

```diff
@@ -0,0 +1,115 @@
+//! Binary plugins support.
+
+use crate::plugin::DynPlugin;
+use async_ffi::FfiFuture;
+use libloading::Library;
+use tower::BoxError;
+
+/// Declare a binary plugin.
+#[macro_export]
+macro_rules! declare_plugin {
+    ($constructor:path, $conf:ty) => {
+        #[no_mangle]
+        pub extern "C" fn _plugin_create(
+            cfg: Box<serde_json::Value>,
+        ) -> FfiFuture<*mut Result<Box<dyn DynPlugin>, BoxError>> {
+            let _subscriber = tracing_subscriber::fmt::init();
```

Comment:
This is because `tracing` doesn't properly support shared objects yet. This will need a better fix than this, but this is good enough for POC.